components/search/ApplyFiltersJS.tsx

```diff
@@ -0,0 +1,68 @@
+import { selectedFilters } from "$store/components/search/SelectedFilters.tsx";
+import { useEffect } from "preact/hooks";
+
+export interface Props {
+  rootId: string;
+  buttonId: string;
+}
+
+function buildParams() {
+  const url = new URL(window.location.href);
+  const existingParams = new URLSearchParams(url.search);
+  const keysToRemove = [];
+  const qValue = existingParams.get('q');
+  if (qValue) {
+    existingParams.set('q', qValue);
+  }
+
+  for (const [key, value] of existingParams.entries()) {
+    if (
+      key !== 'q' &&
+      !selectedFilters.peek().some((filter) =>
+        filter.label === value && `type_tags[${filter.type}][]` === key
+      )
+    ) {
+      keysToRemove.push(key);
+    }
+  }
+
+  keysToRemove.forEach((key) => {
+    existingParams.delete(key);
```

Comment:
inves de criar um array novo e ir incrementando ele voce nao podia colocar o `existingParams.delete` direto no for?
```suggestion
      existingParams.delete(key);
    }
  }
```